Engine/source/T3D/shapeBase.cpp

```diff
@@ -3600,6 +3637,38 @@ DefineEngineMethod( ShapeBase, playThread, bool, ( S32 slot, const char* name ),
    return false;
 }
 
+DefineEngineMethod( ShapeBase, getThreadAnim, const char*, ( S32 slot),,
+   "@brief Get the animation on this thread\n\n"
+
+   "@param slot thread slot to get the animation for. Valid range is 0 - (getMaxThreadCount()-1)\n"  // 3 = ShapeBase::MaxScriptThreads-1
+   "@return Animation name if there is an animation on this thread else empty string.\n\n"
+   
+   "@see pauseThread()\n"
+   "@see stopThread()\n"
+   "@see setThreadDir()\n"
+   "@see setThreadTimeScale()\n"
+   "@see destroyThread()\n")
+{
+   if (slot >= 0 && slot < ShapeBase::MaxScriptThreads) {
+	   return object->getThreadSequenceName(slot);
+   }
+   return "";
+}
+
+DefineEngineFunction( getMaxThreadCount, S32, (),,
+   "@brief Get the max number of threads for ShapeBase objects to animate on\n\n"
+
+   "@return The max number of threads (0 to ShapeBase::MaxScriptThreads-1).\n\n"
+   
+   "@see pauseThread()\n"
+   "@see stopThread()\n"
+   "@see setThreadDir()\n"
+   "@see setThreadTimeScale()\n"
+   "@see destroyThread()\n")
+{
+   return ShapeBase::MaxScriptThreads;
```

Comment:
Seems a bit odd having a function which returns a constant which could just be set as a console variable